storage/innobase/log/log0online.cc

```diff
@@ -900,26 +900,17 @@ void log_online_read_init(void) {
           ? ut_align(log_bmp_sys_unaligned, INNODB_LOG_WRITE_AHEAD_SIZE_MAX)
           : log_bmp_sys_unaligned) log_bitmap_struct;
 
-  /* Initialize bitmap file directory from srv_data_home and add a path
-  separator if needed.  */
-  const size_t srv_data_home_len = strlen(srv_data_home);
-  ut_a(srv_data_home_len < FN_REFLEN);
-  strcpy(log_bmp_sys->bmp_file_home, srv_data_home);
-  if (srv_data_home_len &&
-      log_bmp_sys->bmp_file_home[srv_data_home_len - 1] != SRV_PATH_SEPARATOR) {
-    ut_a(srv_data_home_len < FN_REFLEN - 1);
-    log_bmp_sys->bmp_file_home[srv_data_home_len] = SRV_PATH_SEPARATOR;
-    log_bmp_sys->bmp_file_home[srv_data_home_len + 1] = '\0';
-  }
+  /* Initialize bitmap file directory with a full path to srv_data_home. */
+  log_bmp_sys->bmp_file_home = Fil_path::get_real_path(srv_data_home);
```

Comment:
I don't understand this full path requirement. We should be able to open the file with the relative pathname also "./ib_modified_log*.xb" isn't it?

Reply:
Sorry Satya, I originally found it while testing a different issue so confused a root cause a bit... I've updated fix and description.

Reply:
Thanks for fixing the description and commit message, it is better now